local-env-setup.sh

```diff
@@ -0,0 +1,125 @@
+            export GOPATH=${PWD}/sdks/go/examples/.gogradle/project_gopath && echo "GOPATH was set for this session to '$GOPATH'. Make sure to add this path to your ~/.bashrc file after the execution of this script."
+        fi
+}
+
+# Running on Linux
+if [ "$(uname -s)" = "Linux" ]; then
+    # Assuming Debian based Linux and the prerequisites in https://beam.apache.org/contribute/ are met:
+    apt-get update
+
+    echo "Installing openjdk-8-jdk, python-setuptools, python3-pip, tox, docker.io"
+    apt-get install -y openjdk-8-jdk python-setuptools python3-pip tox docker.io # using python3-pip as Python3 is required.
+
+    type -P python3 > /dev/null 2>&1
+    python3Exists=$?
+    type -P pip3 > /dev/null 2>&1
+    pip3Exists=$?
+    if [ $python3Exists -eq 0  -a $pip3Exists -eq 0 ]; then
+        echo "Installing grpcio-tools mypy-protobuf and virtualenv"
+        pip3 install grpcio-tools mypy-protobuf virtualenv
+    else
+        echo "Python3 and pip3 are required but failed to install. Install them manually and rerun the script."
+        exit
+    fi
+
+    type -P go > /dev/null 2>&1
+    goExists=$?
+    if [ $goExists -eq 0 ]; then
+        install_go_packages
+    else
+        echo "Go is required. Install it manually and rerun the script."
+        exit
+    fi
+fi
+
+# Running on Mac
+if [ "$(uname -s)" = "Darwin" ]; then
+    # Check for Homebrew, install if we don't have it
+    type -P brew > /dev/null 2>&1
+    brewExists=$?
+    if [ $brewExists -ne 0 ]; then
+        echo "Installing homebrew"
+        /bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"
+    fi
+
+    # Update homebrew recipes
+    echo "Updating brew"
+    brew update
+
+    # Assuming we are using brew
+    if brew ls --versions openjdk@8 > /dev/null; then
+        echo "openjdk@8 already installed. Skipping"
+    else
+        echo "Installing openjdk@8"
+        brew install openjdk@8
+    fi
+
+    type -P python3 > /dev/null 2>&1
+    python3Exists=$?
+    type -P pip3 > /dev/null 2>&1
+    pip3Exists=$?
+    if [ $python3Exists -eq 0  -a $pip3Exists -eq 0 ]; then
+        darwin_install_pip3_packages
+    else
+        echo "Python3 and pip3 are required but failed to install. Install them manually and rerun the script."
+        exit
+    fi
+
+    type -P tox > /dev/null 2>&1
+    toxExists=$?
+    if [ $toxExists -eq 0 ]; then
+        echo "tox already installed. Skipping"
+    else
+        echo "Installing tox"
+        brew install tox
+    fi
+
+    type -P docker > /dev/null 2>&1
+    dockerExists=$?
+    if [ $dockerExists -eq 0 ]; then
+        echo "docker already installed. Skipping"
+    else
+        echo "Installing docker"
+        brew install --cask docker
+    fi
+
+    type -P go > /dev/null 2>&1
+    goExists=$?
+    if [ $goExists -eq 0 ]; then
+        install_go_packages
+    else
+        echo "Go is required. Install it manually and rerun the script."
```

Comment:
Consider adding the link to the install docs: https://golang.org/doc/install

Reply:
Thanks, I will continue the work from here.
Solved in https://github.com/apache/beam/pull/14584